turbopack/crates/turbo-tasks/src/task/function.rs

```diff
@@ -267,37 +274,10 @@ macro_rules! task_fn_impl {
             fn functor(&self, this: RawVc, arg: &dyn MagicAny) -> Result<NativeTaskFuture> {
                 let task_fn = self.clone();
                 let recv = Vc::<Recv>::from(this);
-
-                let ($($arg,)*) = get_args::<($($arg,)*)>(arg)?;
-                $(
-                    let $arg = $arg.clone();
-                )*
-
-                Ok(Box::pin(async move {
-                    Output::try_into_raw_vc((task_fn)(recv, $($arg,)*))
-                }))
-            }
-        }
-
-        impl<F, Output, Recv, $($arg,)*> TaskFnInputFunctionWithThis<OperationMode, Recv, ($($arg,)*)> for F
-        where
-            Recv: Sync + Send + 'static,
-            $($arg: TaskInput + 'static,)*
-            F: Fn(OperationVc<Recv>, $($arg,)*) -> Output + Send + Sync + Clone + 'static,
-            Output: TaskOutput + 'static,
-        {
-            #[allow(non_snake_case)]
-            fn functor(&self, this: RawVc, arg: &dyn MagicAny) -> Result<NativeTaskFuture> {
-                let task_fn = self.clone();
-                let recv = OperationVc::<Recv>::from(this);
-
-                let ($($arg,)*) = get_args::<($($arg,)*)>(arg)?;
-                $(
-                    let $arg = $arg.clone();
-                )*
-
+                let ($($arg,)*) = $get_args_fn::<$($arg,)*>(arg)?;
                 Ok(Box::pin(async move {
-                    Output::try_into_raw_vc((task_fn)(recv, $($arg,)*))
+                    let output = (task_fn)(recv, $($arg,)*);
+                    output_try_into_non_local_raw_vc(output).await
                 }))
             }
         }
```

Comment:
I ended up not implementing support for methods with `#[turbo_tasks::function(operation)]`, given that there's a limited number of callsites, and dynamic dispatch likely couldn't work with methods taking `self: OperationVc` anyways.